### PR TITLE
fix: add ability to use working proposals api implementation #LSC-330

### PR DESCRIPTION
**Pull request by NeverHappened**

### CODE REVIEW

**Review by oopcode — APPROVED** [review]

File: proposals/repository.go [hunk 1]
```diff
@@ -53,26 +57,17 @@ func (r *Repository) FetchVoting(ctx context.Context) ([]*models.GetProposalList
...
-		return nil, fmt.Errorf("failed to validate proposals with {%s} status response: %w", status, err)
-	}
+// FetchVoting2 fetches proposals with voting status
+func (r *Repository) FetchVoting2(ctx context.Context) ([]*query.ProposalsOKBodyProposalsItems0, error) {
+	return r.fetch2(ctx, VotingStatus2)
+}
 
-	return resp.GetPayload().Proposals, nil
+// FetchDeposit2 fetches proposals with deposit status
+func (r *Repository) FetchDeposit2(ctx context.Context) ([]*query.ProposalsOKBodyProposalsItems0, error) {
```
Comment (sotnikov-s) [inline proposals/repository.go:69]: don't really like these Fetch...1 and Fetch...2 — the difference is very concealed. could you please expose the differences? ideally reflect the difference in both method names and explicitly in comments

I mean you've preferred one over another during the task, right? the reason for that could imply a good description

File: proposals/repository.go [hunk 2]
```diff
@@ -103,5 +98,59 @@ func (r *Repository) GetVotingProposalVotes(ctx context.Context, proposalID int)
...
+	err = resp.GetPayload().Validate(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate proposals with {%s} status response: %w", status, err)
+	}
+
+	return resp.GetPayload().Proposals, nil
+}
+
+// Uses the more correct lcd API
+// TODO: check that return types correlate with swagger types (query.ProposalsOKBodyProposalsItems0)
```
Comment (sotnikov-s) [inline proposals/repository.go:123]: would be great to have tests for the new methods

File: proposals/repository.go [hunk 3]
```diff
@@ -103,5 +98,59 @@ func (r *Repository) GetVotingProposalVotes(ctx context.Context, proposalID int)
...
+		params := query.ProposalsParams{
+			PaginationKey:   &paginationKey,
+			PaginationLimit: &proposalPageSize,
+			ProposalStatus:  &status,
+			Context:         ctx,
+		}
+
+		resp, err := r.apiClient.Query.Proposals(&params)
+		if err != nil {
+			return res, fmt.Errorf("failed to get proposals with {%s} status: %s\"", err, status)
```
Comment (sotnikov-s) [inline proposals/repository.go:139]: `\"` seems to be redundant
`%w` instead of `%s`
return nil, not res

**Review by sotnikov-s — CHANGES_REQUESTED** [review]

File: proposals/repository.go (hunk 1 above)
Comment (sotnikov-s) [reply proposals/repository.go:69]: or, as swelf suggested in another PR, we can just get rid of the old ones :) as far as I can see, it's only governance bot and proposal checker that uses the repo so far

**Review by foxpy — COMMENTED** [review]
Can't really say much besides API functions suffixed with `2`, that's not cool.

File: proposals/repository_test.go [hunk 4]
```diff
@@ -167,35 +174,39 @@ func validateDepositProposals(t *testing.T, proposals []*models.GetProposalListR
...
+
+	if assert.NotNilf(t, proposal.Content, "proposal content must not be empty") {
+		assert.NotNilf(t, proposal.Content.AtType, "proposal atType must not be empty")
+		assert.NotNilf(t, proposal.Content.Title, "proposal title must not be empty")
+		assert.NotNilf(t, proposal.Content.Description, "proposal description must not be empty")
+		//assert.NotNilf(t, proposal.Content.Recipient, "proposal recipient must not be empty")
+		//if assert.NotNilf(t, proposal.Content.Amount, "proposal amount must not be empty") {
+		//	assert.NotNilf(t, proposal.Content.Amount[0].Denom, "proposal content denom must not be empty")
+		//	assert.NotNilf(t, proposal.Content.Amount[0].Amount, "proposal content amount must not be empty")
+		//}
```
Comment (sotnikov-s) [inline proposals/repository_test.go:188]: why keep commented lines?

File: proposals/repository_test.go [hunk 5]
```diff
@@ -167,35 +174,39 @@ func validateDepositProposals(t *testing.T, proposals []*models.GetProposalListR
...
+		assert.NotNilf(t, proposal.FinalTallyResult.Abstain, "proposal final tally abstain result must not be empty")
+		assert.NotNilf(t, proposal.FinalTallyResult.NoWithVeto, "proposal final tally noWithVeto result must not be empty")
 	}
+
+	assert.NotNilf(t, proposal.SubmitTime, "proposal submitTime must not be empty")
+	assert.NotNilf(t, proposal.DepositEndTime, "proposal depositEndTime must not be empty")
+
+	if assert.NotNilf(t, proposal.TotalDeposit, "proposal totalDeposit must not be empty") {
+		assert.NotNilf(t, proposal.TotalDeposit[0].Denom, "proposal content denom must not be empty")
+		assert.NotNilf(t, proposal.TotalDeposit[0].Amount, "proposal content amount must not be empty")
```
Comment (sotnikov-s) [inline proposals/repository_test.go:205]: proposal `content`->`total deposit` amount. same for denom above

**Review by sotnikov-s — APPROVED** [review]

**Review by foxpy — APPROVED** [review]

**Pull request merged**